### PR TITLE
Bump Version to 0.10.0.dev0

### PR DESCRIPTION
**Pull request by KuuCi**
Working Reg test:
https://databricks.slack.com/archives/C05T1A4UMT8/p1717616549129609

This PR not only bumps the version of LLM-Foundry, but also deprecates top-level variables from yamls. We no longer warn the user but outright throw an error.

### CODE REVIEW

File: llmfoundry/utils/config_utils.py [hunk 1]
```diff
@@ -289,8 +289,8 @@ def make_dataclass_and_log_config(
         unstructured_config['variables'] = {}
 
     for key in extraneous_keys:
-        warnings.warn(
-            f'Unused parameter {key} found in cfg. Please check your yaml to ensure this parameter is necessary. Interpreting {key} as a variable for logging purposes. Top-level variables are deprecated and will not be supported in future releases. Please place any variables under the `variables` key.',
+        raise ValueError(
+            f'Unused parameter {key} found in cfg. Please check your yaml to ensure this parameter is necessary. Please place any variables under the `variables` key.',
             category=DeprecationWarning,
```
Comment (milocress) [inline llmfoundry/utils/config_utils.py:294]:
```suggestion
```

Comment (milocress) [reply llmfoundry/utils/config_utils.py:294]: Nit

File: llmfoundry/utils/config_utils.py [hunk 2]
```diff
@@ -289,8 +289,8 @@ def make_dataclass_and_log_config(
...
         )
         unstructured_config['variables'][key] = unstructured_config.pop(key)
```
Comment (milocress) [inline llmfoundry/utils/config_utils.py:296]:
```suggestion
```

**Review by snarayan21 — DISMISSED** [review]
changes look fine, can someone update the PR description and title to accurately reflect what's being done in the PR

**Comment by snarayan21** [comment]
I'd also want @dakinggg to sign off

**Review by mvpatel2000 — COMMENTED** [review]
why are there changes in the bump version PR?

**Comment by dakinggg** [comment]
@mvpatel2000 we do deprecations in the version bump pr. but also we're gonna wait for the composer bump pr first

File: llmfoundry/utils/config_utils.py [hunk 3]
```diff
@@ -230,6 +236,24 @@ def to_container(
         return cfg  # type: ignore (dicts and lists are already in the correct format)
 
 
+def move_variables_to_section(cfg: DictConfig) -> DictConfig:
```
Comment (dakinggg) [inline llmfoundry/utils/config_utils.py:239]: rm

**Comment by KuuCi** [comment]
> @mvpatel2000 we do deprecations in the version bump pr. but also we're gonna wait for the composer bump pr first

But some of these changes are req for the composer bump to pass

**Review by dakinggg — APPROVED** [review]

**Pull request merged**